src/buffer/out/textBuffer.cpp

```diff
@@ -913,49 +977,32 @@ void TextBuffer::Reset()
         const auto TopRowIndex = (GetFirstRowIndex() + TopRow) % currentSize.Y;
 
         // rotate rows until the top row is at index 0
-        for (auto i = 0; i < TopRowIndex; i++)
-        {
-            _storage.emplace_back(std::move(_storage.front()));
-            _storage.erase(_storage.begin());
-        }
-
+        std::rotate(_storage.begin(), _storage.begin() + TopRowIndex, _storage.end());
         _SetFirstRowIndex(0);
 
         // realloc in the Y direction
         // remove rows if we're shrinking
-        while (_storage.size() > static_cast<size_t>(newSize.Y))
-        {
-            _storage.pop_back();
-        }
-        // add rows if we're growing
-        while (_storage.size() < static_cast<size_t>(newSize.Y))
-        {
-            _storage.emplace_back(gsl::narrow_cast<til::CoordType>(_storage.size()), newSize.X, attributes, this);
-        }
+        _storage.resize(allocator.height());
 
         // Now that we've tampered with the row placement, refresh all the row IDs.
         // Also take advantage of the row ID refresh loop to resize the rows in the X dimension
         // and cleanup the UnicodeStorage characters that might fall outside the resized buffer.
-        _RefreshRowIDs(newSize.X);
+        for (auto& it : _storage)
+        {
+            it.Resize(allocator.chars(), allocator.indices(), allocator.width(), attributes);

```

Comment:
I don't actually understand how this works! Resize should take what is already in the buffer and truncate or grow it, right? Doesn't creating a new `BufferAllocator` give us a totally _new_ block of memory that _doesn't_ contain the existing buffer?

Isn't that gonna clear the screen on a resize?